sharding/nodesCoordinator/numberOfShardsComputer_test.go

```diff
@@ -0,0 +1,61 @@
+package nodesCoordinator
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSovereignNumberOfShardsComputer_ComputeNumberOfShards(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should fail with less than 2", func(t *testing.T) {
+		t.Parallel()
+
+		shardsComputer := NewNumberOfShardsWithMetaComputer()
+		eligibleMap := make(map[uint32][]Validator)
+		eligibleMap[0] = []Validator{&validator{}}
+		nodesConfig := &epochNodesConfig{
+			eligibleMap: eligibleMap,
+		}
+
+		nbShards, err := shardsComputer.ComputeNumberOfShards(nodesConfig)
+		require.Equal(t, ErrInvalidNumberOfShards, err)
+		require.Equal(t, uint32(0), nbShards)
+	})
+
+	t.Run("should work with 2", func(t *testing.T) {
+		t.Parallel()
+
+		shardsComputer := NewNumberOfShardsWithMetaComputer()
+		eligibleMap := make(map[uint32][]Validator)
+		eligibleMap[0] = []Validator{&validator{}}
```

Comment:
Can we also add an entry here with shardID = meta?

Reply:
added